### PR TITLE
Add FlattenParamsWrapper

### PR DESCRIPTION
**Pull request by myleott**
Adds a generic Module wrapper that flattens the Module's parameters into a single contiguous Parameter. This can be useful in many places both as a convenience and to improve speed.

### CODE REVIEW

File: fairscale/nn/misc/flatten_params_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Copyright (c) Tongzhou Wang
+# Licensed under the MIT License.
```
Comment (msbaines) [inline fairscale/nn/misc/flatten_params_wrapper.py:2]: You'll need to update the NOTICE file and the LICENSE and REFERENCES section of README.md

**Review by msbaines — COMMENTED** [review]
LGTM. Thanks! Just need to update NOTICE and README.md.

File: fairscale/nn/misc/flatten_params_wrapper.py (hunk 1 above)
Comment (myleott) [reply fairscale/nn/misc/flatten_params_wrapper.py:2]: Done!

**Comment by myleott** [comment]
Ah, looks like some lint issues causing CI to break. Will submit a fix.

Separately, can we make CI run automatically on PRs?

**Pull request merged**